src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs

```diff
@@ -189,7 +191,7 @@ private DiagnosticInfo ToFileReadDiagnostics(Exception e, CommandLineSourceFile
             return diagnosticInfo;
         }
 
-        internal bool PrintErrors(IEnumerable<Diagnostic> diagnostics, TextWriter consoleOutput)
+        protected bool ReportErrors(IEnumerable<Diagnostic> diagnostics, TextWriter consoleOutput, ErrorLogger errorLogger)
```

Comment:
So I spent a bunch of time trying to do this refactoring, but it seems there are overrides in VB compiler layer that explicitly do some non-diagnostic related reporting, such as printing resolved references, to the console output. For such cases, the compiler indeed needs to differentiate the ConsoleOutput stream from other diagnostic logging stream. I am going to give up on this refactoring as it doesn't seem super critical. Probably it is better if someone from compiler team does it anyways.